runtime/src/storage/mkvs/tree/tree.rs

```diff
@@ -44,27 +44,30 @@ impl Options {
     }
 
     /// Commit the options set so far into a newly constructed tree instance.
-    pub fn new(self, read_syncer: Box<dyn ReadSync>) -> Tree {
-        Tree::new(read_syncer, &self)
+    pub fn new(self, read_syncer: Box<dyn ReadSync>, root_type: u8) -> Tree {
+        Tree::new(read_syncer, root_type, &self)
     }
 }
 
 /// A patricia tree-based MKVS implementation.
 pub struct Tree {
     pub(crate) cache: RefCell<Box<LRUCache>>,
+    pub(crate) root_type: u8,
     pub(crate) pending_write_log: BTreeMap<Key, PendingLogEntry>,
     pub(crate) lock: Arc<Mutex<isize>>,
 }
 
 impl Tree {
     /// Construct a new tree instance using the given read syncer and options struct.
-    pub fn new(read_syncer: Box<dyn ReadSync>, opts: &Options) -> Tree {
+    pub fn new(read_syncer: Box<dyn ReadSync>, root_type: u8, opts: &Options) -> Tree {
```

Comment:
Should this make sure that in case a root is passed via options, the type matches?